File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException("The strands must be of the same length.");
        }

        int hammingDistance = 0;

        for (int i = 0; i < firstStrand.Length; i++)
        {
            if (firstStrand[i] != secondStrand[i])
            {
                hammingDistance++;
            }
        }


        return hammingDistance;
    }
}

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("The strands must be of the same length.")
    return len([(x, y) for (x, y) in zip(list(strand_a), list(strand_b)) if x != y])
